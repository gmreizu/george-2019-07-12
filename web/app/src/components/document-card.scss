.document-card {
    border: 1px solid #ccc;
    padding: 24px;
}
