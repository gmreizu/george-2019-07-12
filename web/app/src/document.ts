/** Document represents a document stored on the service. */
export class Document {
    public static compareByTime(a: Document, b: Document) {
        return b.time.getTime() - a.time.getTime()
    }

    constructor(
        public id: string,
        public title: string,
        public size: number,
        public path: string,
        public time: Date = new Date(),
    ) {
    }
}