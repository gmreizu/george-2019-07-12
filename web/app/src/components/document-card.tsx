import * as React from "react";
import { backendBaseURL } from "../api-client";
import { MainContext } from "../context";
import { Document } from "../document";
import { DeleteDocumentAction } from "../document-store";
import { formatBytes } from "../util";
import "./document-card.scss";

interface Props {
    readonly document: Document
}


/** DocumentCard presents a document as a card. */
export class DocumentCard extends React.PureComponent<Props> {
    static contextType = MainContext

    public render = (): JSX.Element => {
        const { document } = this.props

        return (
            <div className="document-card">
                <h3><a href={`${backendBaseURL}${document.path}`} target="_blank" title={document.title}>{document.title}</a></h3>
                <div>{formatBytes(document.size)}</div>
                <button className="document-card__delete" onClick={this.deleteDidClick}>Delete</button>
            </div>
        )
    }

    private deleteDidClick = () => {
        const { broker } = this.context
        const { document } = this.props
        if (window.confirm("Are you sure you want to delete this Document?")) {
            broker.publish(DeleteDocumentAction, document.id)
        }
    }
}