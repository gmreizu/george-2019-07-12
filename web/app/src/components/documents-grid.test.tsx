import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import { Document } from "../document";
import { DocumentCard } from "./document-card";
import { DocumentsGrid } from "./documents-grid";

const testDocuments = [
    new Document("1", "Doc 1", "/uploads/doc1.jpg", 123),
    new Document("2", "Doc 2", "/uploads/doc1.jpg", 234),
]

describe("DocumentGrid", () => {
    it("renders with documents", () => {
        const wrapper = shallow(<DocumentsGrid documents={testDocuments} />)
        expect(wrapper.contains(<DocumentCard document={testDocuments[0]} />)).toEqual(true)
        expect(wrapper.contains(<DocumentCard document={testDocuments[1]} />)).toEqual(true)
    })

    it("renders when no documents passed", () => {
        shallow(<DocumentsGrid documents={[]} />)
    })

    it("renders correctly", () => {
        const tree = renderer
            .create(<DocumentsGrid documents={testDocuments} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

})
