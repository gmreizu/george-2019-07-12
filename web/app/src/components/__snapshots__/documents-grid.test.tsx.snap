// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentGrid renders correctly 1`] = `
<div
  className="document-grid"
>
  <div
    className="document-card"
  >
    <h3>
      <a
        href="http://localhost:3000/uploads/doc1.jpg"
        rel="noopener noreferrer"
        target="_blank"
        title="Doc 1"
      >
        Doc 1
      </a>
    </h3>
    <div>
      123 Bytes
    </div>
    <button
      className="document-card__delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
  <div
    className="document-card"
  >
    <h3>
      <a
        href="http://localhost:3000/uploads/doc1.jpg"
        rel="noopener noreferrer"
        target="_blank"
        title="Doc 2"
      >
        Doc 2
      </a>
    </h3>
    <div>
      234 Bytes
    </div>
    <button
      className="document-card__delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
